daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/ScriptF.scala

```diff
@@ -696,6 +726,32 @@ object ScriptF {
     }
   }
 
+  private def parseQueryViewContractId(
+      ctx: Ctx,
+      v: SValue,
+  ): Either[String, QueryViewContractId] = {
+    def convert(
+        actAs: SValue,
+        interfaceId: SValue,
+        cid: SValue,
+        stackTrace: Option[SValue],
+        continue: SValue,
+    ) =
+      for {
+        actAs <- Converter.toParties(actAs)
+        interfaceId <- Converter.typeRepToIdentifier(interfaceId)
+        cid <- toContractId(cid)
+        stackTrace <- toStackTrace(ctx, stackTrace)
+      } yield QueryViewContractId(actAs, interfaceId, cid, stackTrace, continue)
+    v match {
+      case SRecord(_, _, ArrayList(actAs, interfaceId, cid, continue)) =>
+        convert(actAs, interfaceId, cid, None, continue)
```

Comment:
I don't think we need the first case since we always have the `stackTrace` field, right?

```suggestion
```

Reply:
You are right.
This pattern of handling optional stack-trace is repeated throughout `ScriptF`.
Probably we should simplify all the cases -- but for now I won't add one more!

Reply:
My (unchecked) guess is that the parsers with both cases are for commands that got the `locations` field after they were introduced, so the case without it is for backwards compatibility. Probably better to wait until we have a cohesive story for daml-script compatibility @remyhaemmerle-da

Reply:
Ah yes. good point!